app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -160,6 +165,9 @@ class AccountManager @Inject constructor(db: AppDatabase) {
 
             Log.d(TAG, "updateActiveAccount: saving account with id " + it.id)
             accountDao.insertOrReplace(it)
+
+            // Drafts alert tracker needs to change its Dao source to current user
+            draftsAlert?.updateActiveAccountId(it.id)
```

Comment:
Directly binding like this doesn't feel very Kotlin-y, but I don't know a better way (flows were suggested but those require you to have a lifetime, which DraftsAlert doesn't; it connects its data to various activity lifetimes to fire the alert, but *this* method only adjusts routing and could occur before any of those lifetimes have been registered)